dos/src/main.rs

```diff
@@ -974,14 +978,24 @@ pub mod test {
             .unwrap();
         let nodes_slice = [node];
 
-        let client = Arc::new(ThinClient::new(
-            cluster.entry_point_info.rpc().unwrap(),
-            cluster
-                .entry_point_info
-                .tpu(cluster.connection_cache.protocol())
-                .unwrap(),
-            cluster.connection_cache.clone(),
-        ));
+        let rpc_pubsub_url = format!("ws://{}/", cluster.entry_point_info.rpc_pubsub().unwrap());
+        let rpc_url = format!("http://{}", cluster.entry_point_info.rpc().unwrap());
+
+        let ConnectionCache::Quic(cache) = &*cluster.connection_cache else {
```

Comment:
`&*` looks a bit suspicious

Reply:
Ya tbh i thought so too lol. But it is used in `ThinClient` [here](https://github.com/anza-xyz/agave/blob/adefcbbb43231cf3516ec976df7fe03843aff3f7/client/src/thin_client.rs#L51), [here](https://github.com/anza-xyz/agave/blob/adefcbbb43231cf3516ec976df7fe03843aff3f7/client/src/thin_client.rs#L71), and [here](https://github.com/anza-xyz/agave/blob/adefcbbb43231cf3516ec976df7fe03843aff3f7/client/src/thin_client.rs#L98). 

Also looks like it is a valid way to get a reference to the internal structure in an Arc<T>. See: https://doc.rust-lang.org/beta/std/sync/struct.Condvar.html and https://stackoverflow.com/questions/62651479/understanding-to-access-a-rust-arc